pkg/offline/offline.go

```diff
@@ -217,6 +271,24 @@ func pushHeartbeats(filepath string, hh []heartbeat.Heartbeat) error {
 	return nil
 }
 
+// PopOptions contain options for poppoing heartbeats from the offline queue.
+type PopOptions struct {
+	Limit    int
+	Excluded []string
+}
+
+// Exclude returns true, if heartbeat with passed in id should be excluded.
```

Comment:
We don't need to check exclude again when sending offline heartbeats since it's already checked when creating the heartbeat. We also want to respect the `wakatime.cfg` state at time of the heartbeat's timestamp, where maybe yesterday the user wanted to include a project but today they don't. In that case, the offline heartbeats from yesterday should still be sent.

Reply:
Maybe using exclude here is a little misleading. But here I keep track of all heartbeats, that were already sent during this cli execution.

As a safety measure, every heartbeat will only be sent once in one wakatime-cli execution. Every heartbeat, which was pushed back to the queue will not be sent again in the same wakatime-cli execution. Reason: As pop and push are separated, we might end up in an endless loop trying to sync the same heartbeats over and over again.

Reply:
We don't need to track heartbeats already sent, we just put them at the back of the queue and allow re-sending them or don't put them back into the queue if a heartbeat returned 400.

Reply:
:+1: